iothub/device/src/Pipeline/MqttTransportHandler.cs

```diff
@@ -197,11 +197,11 @@ internal MqttTransportHandler(PipelineContext context, IotHubClientMqttSettings
                 string uri = $"wss://{_hostName}/$iothub/websocket";
                 _mqttClientOptionsBuilder.WithWebSocketServer(uri);
 
-                IWebProxy proxy = _transportSettings.Proxy;
+                IWebProxy proxy = _mqttTransportSettings.Proxy;
```

Comment:
Retrieve the required values from the transport settings passed in through the context